exchanges/exchange.go

```diff
@@ -1143,18 +1143,23 @@ func (b *Base) FormatExchangeKlineInterval(in kline.Interval) string {
 // ValidateKline confirms that the requested pair, asset & interval are
 // supported and/or enabled by the requested exchange.
 func (b *Base) ValidateKline(pair currency.Pair, a asset.Item, interval kline.Interval) error {
-	var err error
-	if b.CurrencyPairs.IsAssetEnabled(a) != nil {
-		err = common.AppendError(err, fmt.Errorf("%w %v", asset.ErrNotEnabled, a))
-	} else if !b.CurrencyPairs.Pairs[a].Enabled.Contains(pair, true) {
-		err = common.AppendError(err, fmt.Errorf("%w in enabled pairs %v", currency.ErrPairNotFound, pair))
+	var errs error
+	if err := b.CurrencyPairs.IsAssetEnabled(a); err != nil {
+		errs = common.AppendError(errs, err)
```

Comment:
suggestion: All these errors can be returned straight away as this will precipitate failures down stream and not have all these if else spaghetts.